runtime/compiler/z/codegen/J9TreeEvaluator.cpp

```diff
@@ -978,6 +981,111 @@ doubleMaxMinHelper(TR::Node *node, TR::CodeGenerator *cg, bool isMaxOp)
    return node->getRegister();
    }
 
+static TR::Register*
+fpMinMaxHelper(TR::Node* node, TR::CodeGenerator* cg, TR::InstOpCode::Mnemonic compareRROp, TR::InstOpCode::S390BranchCondition returnFirstArgCond, TR::InstOpCode::Mnemonic moveRROp)
+   {
+   TR::Node* lhsNode = node->getChild(0);
+   TR::Node* rhsNode = node->getChild(1);
+
+   TR::Register* lhsReg = cg->gprClobberEvaluate(lhsNode);
+   TR::Register* rhsReg = cg->evaluate(rhsNode);
+
+   TR::LabelSymbol* cFlowRegionStart = generateLabelSymbol(cg);
+   TR::LabelSymbol* cFlowRegionEnd = generateLabelSymbol(cg);
+
+   TR::LabelSymbol* swap = generateLabelSymbol(cg);
+   TR::LabelSymbol* equalRegion = generateLabelSymbol(cg);
+
+   generateS390LabelInstruction(cg, TR::InstOpCode::label, node, cFlowRegionStart);
+   cFlowRegionStart->setStartInternalControlFlow();
+
+   generateRREInstruction(cg, compareRROp, node, lhsReg, rhsReg);
+   generateS390BranchInstruction(cg, TR::InstOpCode::BRC, returnFirstArgCond, node, cFlowRegionEnd);
+
+   //Check for NaN operands for float and double
+   //Support float and double +0/-0 comparisons adhering to IEEE 754 standard
+   //Checking if operands are equal, then branching to equalRegion, otherwise fall through for NaN case handling
+   generateS390BranchInstruction(cg, TR::InstOpCode::BRC, TR::InstOpCode::COND_MASK8, node, equalRegion);
```

Comment:
Would this work with +0 and -0 ? Looking at the instruction, resulting condition code COND_MASK8 will be set when both operands are equal (including 0 operands so +0 equals -0). Can you check if your unit test passes with `-Xjit:disableSIMD` ?

Reply:
yup, unit tests pass with COND_MASK8

from the z/Architecture Principles of Operation:

"If both operands are finite numbers, the comparison is algebraic and follows the procedure for BFP subtraction, except that the difference is discarded after setting the condition code, and both operands remain unchanged. If the difference is exactly zero with either sign, the operands are equal; this includes zero operands (so +0 equals -0). If a nonzero difference is positive or negative, the first operand is high or low, respectively."

Reply:
okk, I checked out the instructions you shared. This comparison is intended - It branches to the section where it checks for +0 and -0/